src/components/Reown/SignOracleDataRequest.js

```diff
@@ -47,43 +49,62 @@ export const SignOracleDataRequestData = ({ data }) => (
 export const SignOracleDataRequest = ({ signOracleData }) => {
   const { dapp, data } = signOracleData;
   const dispatch = useDispatch();
-  const navigation = useNavigation();
+  const [showDeclineModal, setShowDeclineModal] = useState(false);
+
+  const onDeclineTransaction = () => {
+    setShowDeclineModal(true);
+  };
+
+  const { navigateBack } = useBackButtonHandler(
+    onDeclineTransaction,
+  );
 
   const onAcceptSignOracleDataRequest = () => {
     // Signal the user has accepted the current request and pass the accepted data.
     dispatch(reownAccept());
-    navigation.goBack();
```

Comment:
We don't need to go back as we will navigate automatically to the PinScreen after accepting the request.

This prevents a weird animation where it would navigate back and then forward